community/kernel/src/main/java/org/neo4j/kernel/impl/store/kvstore/RotationTimerFactory.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.store.kvstore;
+
+import org.neo4j.helpers.Clock;
+
+public class RotationTimerFactory
+{
+    private Clock clock;
+    private long timeoutMillis;
+
+    public RotationTimerFactory( Clock clock, long timeoutMillis )
+    {
+        this.clock = clock;
+        this.timeoutMillis = timeoutMillis;
+    }
+
+    public RotationTimer createTimer()
+    {
+        return new RotationTimer( clock.currentTimeMillis() );
+    }
```

Comment:
That i do not really like because we will need to either expose clock to check if we timedOut already or call factory again to check if time that was just produce by it is still ok.
RotationTimer keeps clocks/milliseconds internally and provides more convenient way. Agree that look maybe a bit too much for what we need now, but later on we can update and reuse that  for other kind of timeouts.